File: demo/lang/javascript/2_webpage_extract.js
#!/usr/bin/env node
/**
title=extract content from webpage
cid=0
pid=0

1. Load web page from url http://xxx
2. Retrieve img element zt-logo.png in html
3. Check img exist >> `必应`

*/

const https = require('https');

https.get('https://cn.bing.com', function(req) {
    let html = ''

    req.on('data', function(data) {
        html += data;
    });
    req.on('end', () => {
        const res = html.match(/<title>(.*?)</);
        if (res.length > 1) {
            console.log(res[1])
        }
    });
});

File: demo/lang/ruby/2_webpage_extract.rb
#!/usr/bin/env ruby

=begin

title=extract content from webpage
cid=0
pid=0

1. Load web page from url http://xxx
2. Retrieve img element zt-logo.png in html
3. Check img exist >> `必应`

=end

require 'uri'
require 'net/http'

uri = URI('https://cn.bing.com')
res = Net::HTTP.get_response(uri)
html = res.body

elem = html.match(/<title>(.*?)</).captures
puts ""+elem[0]